src/services/SendToGovNotifyService.ts

```diff
@@ -0,0 +1,613 @@
+  			this.environmentVariables.getReminderEmailTemplateId(this.logger),
+  			message,
+  			options,
+  			this.environmentVariables.reminderEmailsGovNotifyUrl(),
+  		);
+  		return emailResponse;
+  	} catch (err: any) {
+  		this.logger.error("Failed to send Reminder Email", {
+  			messageCode: MessageCodes.FAILED_TO_SEND_REMINDER_EMAIL,
+  		});
+  		throw new AppError(
+  			HttpCodesEnum.SERVER_ERROR,
+  			"sendReminderEmail - Failed sending Reminder Email",
+  		);
+  	}
+  }
+
+  async sendDynamicReminderEmail(
+  	message: DynamicReminderEmail,
+  ): Promise<EmailResponse> {
+  	this.logger.info("Sending dynamic reminder email");
+
+  	const { GOV_NOTIFY_OPTIONS } = Constants;
+
+  	try {
+  		const options = {
+  			personalisation: {
+  				[GOV_NOTIFY_OPTIONS.FIRST_NAME]: message.firstName,
+  				[GOV_NOTIFY_OPTIONS.LAST_NAME]: message.lastName,
+  				[GOV_NOTIFY_OPTIONS.CHOSEN_PHOTO_ID]: message.documentUsed,
+  			},
+  			reference: message.referenceId,
+  		};
+
+  		const emailResponse = await this.sendGovNotificationEmail(
+  			this.environmentVariables.getDynamicReminderEmailTemplateId(
+  				this.logger,
+  			),
+  			message,
+  			options,
+  			this.environmentVariables.reminderEmailsGovNotifyUrl(),
+  		);
+  		return emailResponse;
+  	} catch (err: any) {
+  		this.logger.error("Failed to send Dynamic Reminder Email", {
+  			messageCode: MessageCodes.FAILED_TO_SEND_REMINDER_EMAIL,
+  		});
+  		throw new AppError(
+  			HttpCodesEnum.SERVER_ERROR,
+  			"sendReminderEmail - Failed sending Dynamic Reminder Email",
+  		);
+  	}
+  }
+
+  async sendF2FYotiEmailedEvent(message: Email): Promise<void> {
+  	const session = await this.f2fService.getSessionById(message.sessionId);
+  	if (session != null) {
+  		const coreEventFields = buildCoreEventFields(
+  			session,
+  			this.environmentVariables.issuer(),
+  			session.clientIpAddress,
+  		);
+  		try {
+  			await this.f2fService.sendToTXMA({
+  				event_name: TxmaEventNames.F2F_YOTI_PDF_EMAILED,
+  				...coreEventFields,
+  				extensions: {
+  					evidence: [
+  						{
+  							txn: session.yotiSessionId || "",
+  						},
+  					],
+  				},
+  				user: {
+  					...coreEventFields.user,
+  					email: message.emailAddress,
+  					govuk_signin_journey_id: session.clientSessionId,
+  				},
+  			});
+  		} catch (error) {
+  			this.logger.error(
+  				"Failed to write TXMA event F2F_YOTI_PDF_EMAILED to SQS queue.",
+  			);
+  		}
+  	} else {
+  		this.logger.error(
+  			"Failed to write TXMA event F2F_YOTI_PDF_EMAILED to SQS queue, session not found for sessionId: ",
+  			message.sessionId,
+  		);
+  	}
+  }
+
+  async sendF2FLetterSentEvent(message: Email, postalAddress: PersonIdentityAddress): Promise<void> {
+  	const session = await this.f2fService.getSessionById(message.sessionId);
+  	if (session != null) {
+  		const coreEventFields = buildCoreEventFields(
+  			session,
+  			this.environmentVariables.issuer(),
```

Comment:
Can see a lot of code duplication to send the TXMA event, if we can extract the code out and pass on the required fields and event name to send the event instead of duplication.